packages/launcher/src/renderer/launcher/developer/PermissionsRequirements.js

```diff
@@ -112,12 +128,42 @@ export default class PermissionsRequirementsView extends Component<
     this.props.onSetPermissions(formattedPermissions)
   }
 
+  onToggle = (
+    key: PermissionKey,
+    newState: PermissionRequirement,
+    host?: string,
+  ) => {
+    this.setState(({ permissionSettings }) => {
+      if (key === 'WEB_REQUEST' && host) {
```

Comment:
Does this work as expected?
Looks like the logic is mutating the existing object before returning it, so I'm not sure React will consider the state has changes, as AFAIK it compares by references, not by values.